openmls/src/group/mls_group/creation.rs

```diff
@@ -149,6 +143,14 @@ impl MlsGroup {
             key_package,
             private_key: private_key.into(),
         };
+
+        // Delete the [`KeyPackage`] and the corresponding private key from the
+        // key store
+        key_package_bundle
+            .key_package
+            .delete(backend)
+            .map_err(|_| WelcomeError::KeyStoreDeletionError)?;
```

Comment:
Can you maybe add a brief test for this? We probably have to tackle deletion more thoroughly at some point, but it would be good to be sure this works.